lib/ansible/modules/network/fortios/fortios_ipv4_policy.py

```diff
@@ -0,0 +1,336 @@
+  returned: only if config changed
+  type: string
+"""
+
+from ansible.module_utils.fortios import fortios_argument_spec, fortios_required_if
+from ansible.module_utils.fortios import backup
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+
+#check for pyFG lib
+try:
+    from pyFG import FortiOS, FortiConfig
+    from pyFG.exceptions import CommandExecutionException, FailedCommit
+    HAS_PYFG=True
+except:
+    HAS_PYFG=False
+
+def main():
+    argument_spec = dict(
+        comment                   = dict(type='str'),
+        id                        = dict(type='str', required=True),
+        src_intf                  = dict(default='any'),
+        dst_intf                  = dict(default='any'),
+        state                     = dict(choices=['present', 'absent'], default='present'),
+        src_addr                  = dict(required=True, type='list'),
+        dst_addr                  = dict(required=True, type='list'),
+        src_addr_negate           = dict(type='bool', default=False),
+        dst_addr_negate           = dict(type='bool', default=False),
+        policy_action             = dict(choices=['accept','deny'], required=True, aliases=['action']),
+        service                   = dict(aliases=['services'], required=True, type='list'),
+        service_negate            = dict(type='bool', default=False),
+        schedule                  = dict(type='str', default='always'),
+        nat                       = dict(type='bool', default=False),
+        fixedport                 = dict(type='bool', default=False),
+        poolname                  = dict(type='str'),
+        av_profile                = dict(type='str'),
+        webfilter_profile         = dict(type='str'),
+        ips_sensor                = dict(type='str'),
+        application_list          = dict(type='str'),
+    )
+
+    #merge global required_if & argument_spec from module_utils/fortios.py
+    argument_spec.update(fortios_argument_spec)
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=fortios_required_if,
+    )
+
+    #test params
+    #Nat related
+    if not module.params['nat']:
+        if module.params['poolname']:
+            module.fail_json(msg='Poolname param require Nat to be true.')
+        if module.params['fixedport']:
+            module.fail_json(msg='Fixedport param require Nat to be true.')
+
+
+    result = dict(changed=False)
+
+    # fail if libs not present
+    if not HAS_PYFG:
+        module.fail_json(msg='Could not import the python library pyFG required by this module')
+
+    #define device
+    f = FortiOS( module.params['host'],
+        username=module.params['username'],
+        password=module.params['password'],
+        timeout=module.params['username'],
+        vdom=module.params['vdom'])
+
+    path = 'firewall policy'
+
+    #connect
```

Comment:
There is some code duplicated between this pull-request `fortios_ipv4_policy` and other FortisOS pull-requests ([`fortios_address_group`](#21831) and [`fortios_address`](#21542)):

- block calling `connect` method
- block calling `load_config` method

These blocks must be moved together in [`module_utils/fortios.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/fortios.py).

Either an helper method or a parent class (see for example [cloudstack](https://github.com/ansible/ansible/tree/devel/lib/ansible/modules/cloud/cloudstack) modules) must be used instead of duplicating the common code.